src/Model/Contact.php

```diff
@@ -2757,12 +2757,14 @@ public static function addRelationship(array $importer, array $contact, array $d
 		return null;
 	}
 
-	public static function removeFollower($importer, $contact)
+	public static function removeFollower(array $contact)
 	{
-		if (($contact['rel'] == self::FRIEND) || ($contact['rel'] == self::SHARING)) {
+		if (!empty($contact['rel']) && (($contact['rel'] == self::FRIEND) || ($contact['rel'] == self::SHARING))) {
 			DBA::update('contact', ['rel' => self::SHARING], ['id' => $contact['id']]);
-		} else {
+		} elseif (!empty($contact['id'])) {
 			self::remove($contact['id']);
+		} else {
+			DI::logger()->info('Couldn\'t remove follower because of invalid contact array', ['contact' => $contact]);
```

Comment:
Shouldn't we include a callstack in the logger call as well?

Reply:
I guess this would be a good idea, since this function is called by several places.

Reply:
@nupplaphil I don't see any change on this line, did you resolve the conversation a little too fast?

Reply:
.. yes .. Don't change things at hours you're no more fully concentrated ..